test/jdk/javax/swing/plaf/nimbus/TestNimbusJTextPaneColor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+ * @test
+ * @bug 8058704
+ * @key headful
+ * @summary  Verifies if Nimbus honor JTextPane background color
+ * @run main TestNimbusJTextPaneColor
+ */
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Robot;
+import javax.swing.JFrame;
+import javax.swing.JTextPane;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+public class TestNimbusJTextPaneColor {
+
+    static JFrame frame;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                try {
+                    UIManager.setLookAndFeel("javax.swing.plaf.nimbus.NimbusLookAndFeel");
+                } catch (Exception checkedExceptionsPleaseDie) {
+                    throw new RuntimeException(checkedExceptionsPleaseDie);
+                }
+                JTextPane tp = new JTextPane();
+                tp.setForeground(Color.WHITE);
+                tp.setBackground(Color.BLACK);
```

Comment:
There's a brief flash of white background before the actual and proper background color is set. Since the functionality of JTextPane background is rectified/restored, I have raised the PR and I will work on the initial white flashing in a subsequent PR.